src/app/public/modules/angular-tree/angular-tree-node-wrapper.component.ts

```diff
@@ -0,0 +1,160 @@
+})
+export class SkyAngularTreeNodeWrapperComponent implements OnInit {
+
+  @Input()
+  public index: number;
+
+  @Input()
+  public node: TreeNode;
+
+  @Input()
+  public templates: any;
+
+  public set tabIndex(value: number) {
+    this._tabIndex = value;
+  }
+
+  public get tabIndex(): number {
+    return this._tabIndex;
+  }
+
+  public set focused(value: boolean) {
+    this.tabIndex = value ? 0 : -1;
+    if (value) {
+      this.nodeContentWrapperRef.nativeElement.focus();
+    }
+  }
+
+  public set isPartiallySelected(value: boolean) {
+    if (value !== this._isPartiallySelected) {
+      this._isPartiallySelected = value;
+      this.changeDetectorRef.markForCheck();
+    }
+  }
+
+  public get isPartiallySelected(): boolean {
+    return this._isPartiallySelected;
+  }
+
+  public set isSelected(value: boolean) {
+    if (value !== this._isSelected) {
+      this._isSelected = value;
+      this.changeDetectorRef.markForCheck();
+    }
+  }
+
+  public get isSelected(): boolean {
+    return this._isSelected;
+  }
+
+  @ViewChild('nodeContentWrapper')
+  public nodeContentWrapperRef: ElementRef;
+
+  private _isPartiallySelected: boolean;
+
+  private _isSelected: boolean;
+
+  private _tabIndex: number = -1;
+
+  constructor(
+    private changeDetectorRef: ChangeDetectorRef,
+    @Optional() private skyAngularTreeWrapper: SkyAngularTreeWrapperComponent
+  ) { }
+
+  public ngOnInit(): void {
+    // Because we're binding the checkbox to node's children properties, we need to manually control change detection.
+    // Here, we listen to the tree's state and force change detection in the setters if the value has changed.
+    this.node.treeModel.subscribeToState((state: ITreeState) => {
+      this.isSelected = this.node.isSelected;
+      this.isPartiallySelected = this.node.isPartiallySelected;
+
+      if (state.focusedNodeId) {
+        this.focused = state.focusedNodeId === this.node.id;
+      }
+    });
+
+    // On init, make first root node tabbable.
+    if (this.node.isRoot && this.node.index === 0) {
+      this.tabIndex = 0;
+    }
+
+    if (!this.skyAngularTreeWrapper) {
+      console.warn(`<sky-angular-tree-node-wrapper> should be wrapped inside a <sky-angular-tree-wrapper> component.`);
+    }
+  }
+
+  public onCheckboxChange(node: TreeNode, event: SkyCheckboxChange): void {
+    this.toggleSelected(node, node.treeModel, event);
+    node.setIsActive(event.checked);
+  }
+
+  public onNodeContentClick(node: TreeNode, event: any): void {
+    if (this.node.options.useCheckbox && this.isSelectable()) {
+      this.toggleSelected(node, node.treeModel, event);
+    }
+    this.node.mouseAction('click', event);
+  }
+
+  public showCheckbox(): boolean {
+    // Check for checkbox mode enabled, but also respect leaf-node and single-select settings.
+    return this.node.options.useCheckbox && this.isSelectable() && !this.skyAngularTreeWrapper.selectSingle;
```

Comment:
I could make the same argument for `this.node.options.useCheckbox` being first, since that is the property the consumer has to turn on to enable checkboxes.

Reply:
Sounds good